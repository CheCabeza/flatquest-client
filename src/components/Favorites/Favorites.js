import * as React from 'react';
import {View, Image, StyleSheet, Text, Pressable} from 'react-native';
import {connect} from 'react-redux';
import {PropTypes} from 'prop-types';

function Favorites({navigation, flats, dispatch, user}) {
  return (
    <View style={styles.page}>
      <Text style={styles.favText}>Favorites</Text>
      <View style={styles.favSection}>
        {flats
          .filter(flat => user.user.favorites.includes(flat._id))
          .map((flatsData, index) => (
            <Pressable
              key={index}
              onPress={() =>
                navigation.navigate('Details', {flatId: flatsData._id})
              }>
              <View style={styles.box} key={flatsData.index}>
                <Image
                  testID="detailButton"
                  style={styles.img}
                  source={{
                    uri: `${flatsData.photos[0]}`,
                  }}
                />
              </View>
            </Pressable>
          ))}
      </View>
    </View>
  );
}

Favorites.propTypes = {
  flats: PropTypes.array.isRequired,
  dispatch: PropTypes.func.isRequired,
};

function mapStateToProps(store) {
  return {
    flats: store.flats,
    user: store.user,
  };
}

export default connect(mapStateToProps)(Favorites);

const styles = StyleSheet.create({
  page: {
    flex: 1,
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'column',
    backgroundColor: 'white',
  },
  favText: {
    fontSize: 20,
    marginTop: 70,
    marginBottom: 10,
  },
  img: {
    borderRadius: 20,
    width: 170,
    height: 120,
  },
  box: {
    shadowColor: '#000',
    margin: 10,
    shadowOffset: {
      width: 0,
      height: 6,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  favSection: {
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
});
