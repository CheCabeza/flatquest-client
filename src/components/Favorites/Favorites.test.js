import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import {useNavigation} from '@react-navigation/native';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import Favorites from './Favorites';
import thunk from 'redux-thunk';

jest.mock('@react-navigation/native');

describe('Given a Favorites component', () => {
  const mockStore = configureStore([thunk]);
  let store;
  let navigation;

  beforeEach(() => {
    store = mockStore({
      user: {
        user: {
          email: 'ho@la.com',
          pass: '123123123',
          favorites: [123],
        },
      },
      flats: [
        {
          _id: 123,
          photos: 'a',
          pros: ['glass', 'folder-open'],
          cons: ['money', 'copy'],
          reviews: ['Mierda de test'],
        },
      ],
    });
    useNavigation.mockReturnValueOnce({
      navigate: jest.fn(),
    });
    navigation = {
      navigate: jest.fn(),
    };
  });

  test('Should render the Favorites of page', () => {
    const lobby = render(
      <Provider store={store}>
        <Favorites />
      </Provider>,
    );
    expect(lobby).toMatchSnapshot();
  });

  test('Then navigation.navigate is invoked', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <Favorites navigation={navigation} />
      </Provider>,
    );
    const button = getByTestId('detailButton');
    fireEvent.press(button);
    expect(navigation.navigate).toHaveBeenCalled();
  });
});
