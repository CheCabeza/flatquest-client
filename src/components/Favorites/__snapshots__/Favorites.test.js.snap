// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Favorites component Should render the Favorites of page 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "marginBottom": 10,
        "marginTop": 70,
      }
    }
  >
    Favorites
  </Text>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Object {
            "elevation": 5,
            "margin": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "a",
            }
          }
          style={
            Object {
              "borderRadius": 20,
              "height": 120,
              "width": 170,
            }
          }
          testID="detailButton"
        />
      </View>
    </View>
  </View>
</View>
`;
